### ISSUE TITLE
qtPrepareTool qmake function fails to locate perl scripts

### ISSUE THREAD

**Issue by michaelforney** [NixOS/nixpkgs#76080]
**Describe the bug**

The patched `qtPrepareTool` qmake function in the qt5.qtbase package fails to detect perl scripts (specifically `syncqt.pl`), causing a failure when trying to build qtwebengine manually.

**To Reproduce**
1. `nix-shell -p pkg-config qt5Full ffmpeg_4 libwebp libopus xorg.libX11 gperf bison python2 mesa libglvnd nss dbus fontconfig xorg.libXcomposite xorg.libXcursor xorg.libXi xorg.libXtst libjpeg git ninja`
2. `git clone --recursive https://code.qt.io/qt/qtwebengine.git && cd qtwebengine`
3. `qmake && make`

**Expected behavior**
The build completes successfully.

**Additional context**
After building gn, the build fails with the following error:
```
...
cd core/ && ( test -e Makefile || /nix/store/qys53np1vhsgz6qpg7vk8pn56wrjj6kf-qt-full-5.12.6/bin/qmake -o Makefile /home/michael/src/tmp/qtwebengine/src/core/core.pro ) && make -f Makefile 
make[2]: Entering directory '/home/michael/src/tmp/qtwebengine/src/core'
( test -e Makefile.core_headers || /nix/store/qys53np1vhsgz6qpg7vk8pn56wrjj6kf-qt-full-5.12.6/bin/qmake -o Makefile.core_headers /home/michael/src/tmp/qtwebengine/src/core/core_headers.pro ) && make -f Makefile.core_headers 
Project MESSAGE: -module QtWebEngineCore -version 6.0.0 -outdir /home/michael/src/tmp/qtwebengine -builddir /home/michael/src/tmp/qtwebengine /home/michael/src/tmp/qtwebengine
sh: -module: not found
Project ERROR: Failed to run: -module QtWebEngineCore -version 6.0.0 -outdir /home/michael/src/tmp/qtwebengine -builddir /home/michael/src/tmp/qtwebengine /home/michael/src/tmp/qtwebengine
make[2]: *** [Makefile:48: sub-core_headers-pro-make_first] Error 3
make[2]: Leaving directory '/home/michael/src/tmp/qtwebengine/src/core'
make[1]: *** [Makefile:78: sub-core-make_first] Error 2
make[1]: Leaving directory '/home/michael/src/tmp/qtwebengine/src'
make: *** [Makefile:49: sub-src-make_first] Error 2
```

This is because `qtPrepareTool(QMAKE_SYNCQT, syncqt, , system)` is failing to find `syncqt.pl` on the system, and returns an empty string instead. This function is patched in nixpkgs here:
https://github.com/NixOS/nixpkgs/blob/2197fed076cf8b82f8cffeb676e9a025ce33e9b7/pkgs/development/libraries/qt-5/5.12/qtbase.patch.d/0003-qtbase-mkspecs.patch#L372-L384

The original (upstream) version the function checked that `syncqt.pl` existed in the QT_HOST_BINS directory. The patched version checks that `syncqt` exists in the user's path (which it doesn't), and then appends `.pl` to the result.

Instead, perhaps the function should be patched to run `command -v $$2.pl` if the result of `command -v $$2` is empty.

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.3.8+`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.1`
 - channels(michael): `"nixpkgs-20.03pre204216.cc6cf0a96a6"`
 - nixpkgs: `/home/michael/.nix-defexpr/channels/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by OPNA2608** [comment]
Still important, just encountered this while trying to package something for Nixpkgs.

The project loads Qt's `mkspecs/features/qt_module_headers.prf`which attempts to find Qt's own syncqt.pl tool (`${qtbase.dev}/bin/syncqt.pl`) and errors out.

```
Project MESSAGE: -module UbuntuGestures -version 5.5.0 -outdir /build/source -builddir /build/source /build/source
sh: -module: not found
Project ERROR: Failed to run: -module UbuntuGestures -version 5.5.0 -outdir /build/source -builddir /build/source /build/source
```
error location: https://github.com/qt/qtbase/blob/5.15.0/mkspecs/features/qt_module_headers.prf#L28

The intended output of this from a CI run:

```
Project MESSAGE: perl -w /usr/lib/qt5/bin/syncqt.pl -module UbuntuGestures -version 5.5.0 -outdir '/build/ubuntu-ui-toolkit-1.3.2192~0ubports1+0~20201106203828.18~1.gbp62f90d' /build/ubuntu-ui-toolkit-1.3.2192~0ubports1+0~20201106203828.18~1.gbp62f90d
```

Possibly related: The script's interpreter is set to `#!/usr/bin/env perl`, which doesn't seem to work while using nix-build. Testing `syncqt.pl` within `configurePhase` gives:

```
/nix/store/zbdjq217kqq86d5qnn8aiqpf200m6j98-stdenv-linux/setup: /nix/store/cixibhdl3dfmj65w17ip55m16m04gkhw-qtbase-5.15.0-dev/bin/syncqt.pl: /usr/bin/env: bad interpreter: No such file or directory
```

**Comment by OPNA2608** [comment]
Should be fixed by #118904.
